src/fides/api/api/v1/endpoints/pre_approval_webhook_endpoints.py

```diff
@@ -0,0 +1,259 @@
+from loguru import logger
+from pydantic import conlist
+from sqlalchemy.orm import Session
+from starlette.exceptions import HTTPException
+from starlette.status import HTTP_200_OK, HTTP_400_BAD_REQUEST, HTTP_404_NOT_FOUND
+
+from fides.api.api import deps
+from fides.api.common_exceptions import KeyOrNameAlreadyExists
+from fides.api.db.base_class import get_key_from_data
+from fides.api.models.policy import PolicyPreWebhook
+from fides.api.models.pre_approval_webhook import PreApprovalWebhook
+from fides.api.oauth.utils import verify_oauth_client
+from fides.api.schemas import pre_approval_webhooks as schemas
+from fides.api.util.api_router import APIRouter
+from fides.api.util.connection_util import get_connection_config_or_error
+from fides.common.api import scope_registry as scopes
+from fides.common.api.v1 import urn_registry as urls
+
+router = APIRouter(tags=["Pre Approval Webhooks"], prefix=urls.V1_URL_PREFIX)
+
+
+@router.get(
+    urls.WEBHOOK_PRE_APPROVAL,
+    status_code=HTTP_200_OK,
+    response_model=Page[schemas.PreApprovalWebhookResponse],
+    dependencies=[Security(verify_oauth_client, scopes=[scopes.WEBHOOK_READ])],
+)
+def get_pre_approval_webhook_list(
+    *,
+    db: Session = Depends(deps.get_db),
+    params: Params = Depends(),
+) -> AbstractPage[schemas.PreApprovalWebhookResponse]:
+    """
+    Return a paginated list of all PreApprovalWebhook records in this system
+    """
+    logger.info("Finding all pre_approval webhooks with pagination params '{}'", params)
+    pre_approval_webhooks = PreApprovalWebhook.query(db=db).order_by(
+        PreApprovalWebhook.created_at.desc()
+    )
+    return paginate(pre_approval_webhooks, params=params)
+
+
+def get_pre_approval_webhook_or_error(
+    db: Session, webhook_key: FidesKey
+) -> PreApprovalWebhook:
+    """Helper method to load PreApprovalWebhook or throw a 404"""
+    logger.info("Finding PreApprovalWebhook with key '{}'", webhook_key)
+    pre_approval_webhook = PreApprovalWebhook.get_by(
+        db=db, field="key", value=webhook_key
+    )
+    if not pre_approval_webhook:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No Policy found for key {webhook_key}.",
+        )
+
+    return pre_approval_webhook
+
+
+@router.get(
+    urls.WEBHOOK_PRE_APPROVAL_DETAIL,
+    status_code=HTTP_200_OK,
+    response_model=schemas.PreApprovalWebhookResponse,
+    dependencies=[Security(verify_oauth_client, scopes=[scopes.WEBHOOK_READ])],
+)
+def get_pre_approval_webhook_detail(
+    *,
+    db: Session = Depends(deps.get_db),
+    webhook_key: FidesKey,
+) -> PolicyPreWebhook:
+    """
+    Loads the given Pre-Approval Webhook by key
+    """
+    return get_pre_approval_webhook_or_error(db, webhook_key)  # type: ignore
+
+
+def put_webhooks(
+    db: Session = Depends(deps.get_db),
+    webhooks: List[schemas.PreApprovalWebhookCreate] = Body(...),
+) -> List[PreApprovalWebhook]:
+    """
```

Comment:
Ah okay I was wondering why you'd extracted these functions out since they were only being called in one place, but I see you've patterned it off of policy_webhook_endpoints where the corresponding function was shared between multiple endpoints.

Reply:
yeah, I can inline actually!

Reply:
not a big deal either way -